index.js

```diff
@@ -52,4 +52,150 @@ const state = {
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+
+// hint for cart: make a copy of a store item, add a property called
+// quantity
+
+// SELECT EXISTING HTML ELEMENTS
+
+const storeUl = document.querySelector('ul')
+const cartUl = document.querySelector('.cart--item-list-container ul')
+const totalSection = document.querySelector ('.total-section')
+
+//Items in the shop
+function renderStoreItems() {
+ state.items.forEach((item) => {
+  // Create List Item
+  const storeItemli = document.createElement('li')
+  // Store list item in UL
+  storeUl.append(storeItemli)
+  // Create image and save to list item
+  const storeItemImage = document.createElement('img')
+  storeItemImage.setAttribute("src", "assets/icons/" + item.id + ".svg")
+  storeItemli.append(storeItemImage)
+  // Create Button w/ click
+  const addToCartButton = document.createElement('button')
+  addToCartButton.innerText = "Add to cart"
+  storeItemli.append(addToCartButton)
+  addToCartButton.addEventListener('click', function(event){
+    event.preventDefault()
+    addingToCartArray(item)
+  })
+ });
+};
+
+// When button is clicked, pass the item object into this function
+// Is the item already in the array? If yes, increase quantity
+// If not add object into array
+function addingToCartArray(item){
+  const index = state.cart.findIndex((element) => element.id === item.id)
+  // If Element.id (id of the item in the cart array) does not match item.id will return -1 
+  if(index === -1){
+    const cartObject = {
+      id: item.id,
+      name: item.name,
+      price: item.price,
+      quantity: 1
+    }
+    state.cart.push(cartObject)
+    addingItemsToCart()
+    return
+  }
+  state.cart[index].quantity += 1
+  addingItemsToCart()
+
+}  
+
+
+// render Items to Cart from function above
+function addingItemsToCart() {
+  cartUl.innerHTML = ''
+  // Create For Each loop to cycle through array and add them to the list
+  // const index = state.cart.find((element) => element.id)
```

Comment:
clever to carry over the const so that you can remind yourself what is in the scope/not have to scroll up to find it!